pkg/controller/migration/namespace_migration.go

```diff
@@ -672,6 +672,14 @@ func (m *CoreNamespaceMigration) waitUntilNodeCanBeMigrated(ctx context.Context,
 			}
 		}
 
+		// if the cluster has only 1 node then once the ensureTyphaRoom() function runs, it scales down the kube-system/calico-typha pod.
+		// then it can happen that kube-system/calico-node goes down to CrashLoopBackOff due that.
+		// When that happens, the below calculation does not work for 1 node clusters. Handle this special case. Assumptions:
+		// - all the nodes are labeled with projectcalico.org/operator-node-migration (pre-operator value)
+		// - kube-system/calico-node and calico-system/calico-node daemonsets are using the above nodeSelector
+		if ksD == 1 && csD == 0 {
+			ksR = 1
```

Comment:
For this case I think we should just `return true,nil` rather than setting `ksR = 1` which results in the same return.

Reply:
Sure, I agree. Thank you!